presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -667,6 +690,36 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        if (totalSegments == 1) {
+            return getPartitions(databaseName, tableName, expression, null);
+        }
+
+        // Do parallel partition fetch.
+        CompletionService<List<Partition>> completionService = new ExecutorCompletionService<>(executor);
+        IntStream.range(0, totalSegments)
```

Comment:
I think this would be easier to read as a traditional for loop

Reply:
Sure.